File: week15/week15-1.cpp
// week15-1.cpp LeetCode 3158 要找出現次數
int duplicateNumbersXOR(int* nums, int numsSize) {
    int table[51] = {};
    for(int i=0; i<numsSize; i++) {
        table[nums[i]]++;  //這個數字出現， +1
    }
    int ans = 0;
    for(int i=1; i<=50; i++) {  //針對 table裡 1...50的數
        if(table[i]==2) ans = ans ^ i;
    }
    return ans;
}

File: week15/week15-1.py
class Solution:  # week15-1.py LeetCode 3158 要找出現2次
    def duplicateNumbersXOR(self, nums: List[int]) -> int:
        table = {}  #大括號:table[num] 對應的次數
        for num in nums: #每個數字輪一次
            if num in table:  #出現過的話
                table[num] += 1  #次數+1
            else:
                table[num] = 1  #第一次出現
        # print(table)  #答案還沒算出來

        ans = 0
        for num in table:
            if table[num] == 2:
                ans = ans ^ num
        return ans
